src/profile-logic/profile-data.js

```diff
@@ -1081,3 +1083,45 @@ export function getEmptyProfile(): Profile {
     threads: [],
   };
 }
+
+export function getOriginAnnotationForFunc(
```

Comment:
I don't see origin annotation documented anywhere, and I'm not very clear on what the word means. Please add some comments about what this function is doing and what it means. I believe it's existed around in one form another up until this point.